tenseal/binding.cpp

```diff
@@ -50,46 +50,74 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
                     BFVVector::Create(ctx, data));
             }))
         .def("size", py::overload_cast<>(&BFVVector::size, py::const_))
-        .def("decrypt", py::overload_cast<>(&BFVVector::decrypt, py::const_))
-        .def("decrypt", py::overload_cast<const std::shared_ptr<SecretKey> &>(
-                            &BFVVector::decrypt, py::const_))
+        .def("decrypt",
+             [](shared_ptr<BFVVector> obj) { return obj->decrypt().data(); })
+        .def("decrypt",
+             [](shared_ptr<BFVVector> obj,
+                const std::shared_ptr<SecretKey> &sk) {
+                 return obj->decrypt(sk).data();
+             })
         .def("add", &BFVVector::add)
         .def("add_", &BFVVector::add_inplace)
-        .def("add_plain", py::overload_cast<const vector<int64_t> &>(
-                              &BFVVector::add_plain, py::const_))
-        .def("add_plain_", py::overload_cast<const vector<int64_t> &>(
-                               &BFVVector::add_plain_inplace))
+        .def("add_plain",
+             [](shared_ptr<BFVVector> obj, const vector<int64_t> &other) {
+                 return obj->add_plain(other);
+             })
```

Comment:
The operations now require a PlainTensor parameter and pybind doesn't know how to cast python vectors to PlainTensor, although PlainTensor has cast operator for std::vector.
Or I do not know how to indicate the cast to pybind.

Reply:
If you have any suggestion here, I can try it. Not sure what type uses pybind internally for python vector.

But it could be helpful if we expose similar overloads for PyTorch or Tensorflow tensors, right now doesn't seem that useful, true.

Reply:
As far as I know, pybind11 can converts python list into vectors only, but I think we can add custom type casters so that pybind11 knows how to do it.
I only read about it, but it's supposed to work. With this I guess we can only provide custom casters for different types (python lists, torch tensors...) without the need to wrap each function.

Reply:
I think I need to define pybind11::class_ for PlainTensor instantions to make it work, I will try that.

Reply:
I'm also up to leave it for another PR. Just let me know

Reply:
I am unable to make it work.

We need to implement a [PYBIND11_TYPE_CASTER](https://pybind11.readthedocs.io/en/stable/advanced/cast/custom.html) to make PlainTensor compatible with python lists, but I need to research this a bit more.

I'd leave it for another PR.